src/domains.cpp

```diff
@@ -333,15 +308,24 @@ unsigned int check_electric_field (dolfin::Function *voltage,
 
   // mark for refinement
   MeshFunction<bool> cell_marker(mesh, mesh.topology().dim(), false);
+  double new_entropy_tol = entropy_tol;
   unsigned int marked_elem_count = 0;
   for ( uint errVecInd = 0; errVecInd < error_vector.size(); errVecInd++) {
-    if ( error_vector[errVecInd] > entropy_tol/2.0 ) {
-        marked_elem_count++;
-        cell_marker.values()[errVecInd] = true;
+    if (error_vector[errVecInd] > new_entropy_tol) {
+      marked_elem_count++;
+      cell_marker.values()[errVecInd] = true;
+      
+      if (marked_elem_count > max_elements) {
+        cell_marker.values()[errVecInd] = false;
+        new_entropy_tol *= 5.0;
+        errVecInd = 0;
```

Comment:
that wouldn't put us back to the top of the for loop though... This makes the for-loop start over.